addons/web/static/tests/views/fields/properties_field_tests.js

```diff
@@ -984,6 +984,164 @@ QUnit.module("Fields", (hooks) => {
         );
     });
 
+    /**
+     * When the user creates a property field of type many2many, many2one, etc.
+     * and changes the co-model of the field, the model loaded by the "Search more..."
+     * modal should correspond to the selected model and should be updated dynamically.
+     */
+    QUnit.test("properties: many2many 'Search more...' loaded model", async function (assert) {
+        async function mockRPC(route, { method, model }) {
+            if (method === "check_access_rights") {
+                return true;
+            }
+            if (method === "name_search") {
+                if (model === "res.partner") {
+                    return [
+                        [1, "Partner 1"],
+                        [2, "Partner 2"],
+                        [3, "Partner 3"],
+                        [4, "Partner 4"],
+                        [5, "Partner 5"],
+                        [6, "Partner 6"],
+                        [7, "Partner 7"],
+                        [8, "Partner 8"]
+                    ];
+                }
+                if (model === "res.users") {
+                    return [
+                        [1, "User 1"],
+                        [2, "User 2"],
+                        [3, "User 3"],
+                        [4, "User 4"],
+                        [5, "User 5"],
+                        [6, "User 6"],
+                        [7, "User 7"],
+                        [8, "User 8"]
+                    ];
+                }
+            }
+            if (method === "display_name_for" && model === "ir.model") {
+                return [
+                    { display_name: "Partner", model: "res.partner" },
+                    { display_name: "User", model: "res.users" },
+                ];
+            }
+            if (method === "get_available_models" && model === "ir.model") {
+                return [
+                    { model: "res.partner", display_name: "Partner" },
+                    { model: "res.users", display_name: "User" },
+                ];
+            }
+        }
+        await makeView({
+            type: "form",
+            resModel: "res.partner",
+            resId: 1,
+            serverData: {
```

Comment:
Can't we complete the serverData already used by these tests instead? Seems tedious to re-specify everything.

Reply:
(See other comment)